Source/Core/GeometryPipeline.js

```diff
@@ -2204,23 +2206,53 @@ define([
         var length = positions.length / 3;
         for (i = 0; i < length; i += 4) {
             var i0 = i;
-            var i1 = i + 1;
             var i2 = i + 2;
-            var i3 = i + 3;
 
             var p0 = Cartesian3.fromArray(positions, i0 * 3, cartesian3Scratch0);
-            var p1 = Cartesian3.fromArray(positions, i1 * 3, cartesian3Scratch1);
             var p2 = Cartesian3.fromArray(positions, i2 * 3, cartesian3Scratch2);
-            var p3 = Cartesian3.fromArray(positions, i3 * 3, cartesian3Scratch3);
 
-            if (Math.abs(p0.y) < CesiumMath.EPSILON6) {
-                p0.y = CesiumMath.EPSILON6 * (p2.y < 0.0 ? -1.0 : 1.0);
-                p1.y = p0.y;
+            if (Math.abs(p0.y) < coplanarOffset) {
+                if (p2.y < 0.0) {
+                    p0.y = -coplanarOffset;
+
+                    positions[i * 3 + 1] = p0.y;
+                    positions[(i + 1) * 3 + 1] = p0.y;
+
+                    for (j = i0 * 3; j < i0 * 3 + 4 * 3; ++j) {
+                        prevPositions[j] = positions[j];
```

Comment:
Shouldn't `positions` and `prevPositions` be offset by 3?
Also, the only values changing are `positions[i * 3 + 1]` and `positions[(i + 1) * 3 + 1]`.  You can just update the corresponding prevPositions instead of doing this in a loop.  

Same for the else statement below